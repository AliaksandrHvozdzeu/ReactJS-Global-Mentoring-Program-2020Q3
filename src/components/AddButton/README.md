#Add movie button component
![image](https://drive.google.com/uc?export=view&id=1Q_9Bx7j59VMrDbfCyjMurZMBA6_ibDbD)
