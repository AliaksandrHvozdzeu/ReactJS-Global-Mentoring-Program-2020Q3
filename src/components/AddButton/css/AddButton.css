.add-button {
    color: var(--red);
    background-color: var(--grey-transparent);
    text-transform: uppercase;
    padding: 1rem 1rem;
    border-radius: 4px;
    font-size: 1rem;
    outline: none;
    position: absolute;
    right: 0;
    top: 1rem;
    font-weight: 600;
}

.add-button:hover {
    background-color: var(--grey-lighter-transparent);
}

.add-button:active {
    background-color: var(--grey);
}
