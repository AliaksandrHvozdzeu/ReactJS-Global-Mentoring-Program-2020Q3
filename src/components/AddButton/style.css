.add-button {
    color: var(--red);
    background-color: var(--grey-transparent);
    text-transform: uppercase;
    padding: 1rem 1rem;
    border-radius: 4px;
    font-size: 1rem;
    outline: none;
    position: absolute;
    right: 0;
    top: 1rem;
}

.add-button:hover {
    background-color: var(--grey-lighter-transparent);
}

.add-button:active {
    background-color: var(--grey);
}

.add-movie-modal {
    width: 45rem;
    position: fixed;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    background: #232323;
    overflow: auto;
    outline: none;
    padding: 52px 52px 30px;
}

.add-movie-modal-overlay {
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: rgba(0, 0, 0, 0.75);
}

.ReactModal__Overlay {
    opacity: 0;
    transition: opacity 500ms ease-in-out;
}

.ReactModal__Overlay--after-open {
    opacity: 1;
    z-index: 10;
}

.ReactModal__Overlay--before-close {
    opacity: 0;
}
