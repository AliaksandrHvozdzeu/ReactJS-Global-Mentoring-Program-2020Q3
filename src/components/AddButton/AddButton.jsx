import React, { useState } from 'react';
import MovieModal from '../MovieModal';
import MovieAdd from '../MovieAdd';
import './style.css';


export default function AddButton() {

  const [isOpen, setIsOpen] = useState(false);

  const openModal = () => {
    setIsOpen(!isOpen);
  };

  const closeModal = () => {
    setIsOpen(false);
  };

  return (
    <div className="App">
      <button type="button" className="add-button" onClick={openModal}>+ ADD MOVIE</button>
      <MovieModal isOpen={isOpen} openModal={openModal} modalContent={<MovieAdd closeAction={closeModal} modalTitle="ADD MOVIE"/>}/>
    </div>
  );
}
