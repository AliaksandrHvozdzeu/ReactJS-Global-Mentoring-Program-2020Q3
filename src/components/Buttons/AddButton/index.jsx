import React from 'react';
import './style.css';

export default () => (
  <>
    <button type="button" className="add-button">+ ADD MOVIE</button>
  </>
);
