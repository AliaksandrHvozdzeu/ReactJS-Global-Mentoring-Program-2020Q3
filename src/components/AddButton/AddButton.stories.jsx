import React from 'react';
import { withKnobs } from '@storybook/addon-knobs';
import AddButton from './js/AddButton';
import '../../App.css';

export default {
  title: 'AddButton',
  decorators: [withKnobs],
};

export const Default = () => (
  <AddButton />
);
