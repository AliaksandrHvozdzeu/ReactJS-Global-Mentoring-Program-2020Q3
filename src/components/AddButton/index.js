import AddButton from './js/AddButton';

export default AddButton;
