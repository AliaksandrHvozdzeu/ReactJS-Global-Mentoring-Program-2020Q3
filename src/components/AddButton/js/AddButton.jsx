import React, { useState } from 'react';
import MovieModal from '../../MovieModal';
import MovieAdd from '../../MovieAdd';
import '../css/AddButton.css';

const AddButton = () => {

  const [isOpen, setIsOpen] = useState(false);

  const openAddModal = () => {
    setIsOpen(!isOpen);
  };

  const closeAddModal = () => {
    setIsOpen(false);
  };

  const modalContent = (
    <MovieAdd closeAction={closeAddModal} modalTitle="ADD MOVIE"/>
  );

  return (
    <>
      <button type="button" className="add-button" onClick={openAddModal}>+ ADD MOVIE</button>
      <MovieModal isOpen={isOpen}
                  openModal={openAddModal}
                  modalContent={modalContent}/>
    </>
  );
};

export default AddButton;
