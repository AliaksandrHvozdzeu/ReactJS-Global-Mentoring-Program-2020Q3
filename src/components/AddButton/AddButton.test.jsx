import React from 'react';
import rerender from 'react-test-renderer';
import AddButton from './js/AddButton';
import { shallow } from "enzyme";

describe('when AddButton', () => {

  test('then snapshot created', () => {
    const component = rerender.create(<AddButton/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('openModal', () => {
    const component = shallow(<AddButton />);
    component.openAddModal;
    const addButton = component.find('.add-button');
    addButton.simulate('click');
    expect(addButton.text()).toEqual("+ ADD MOVIE");
    expect(addButton.hasClass("add-button")).toEqual(true);
  });

  it('closeModal ', () => {
    const component = shallow(<AddButton />);
    component.closeAddModal;
    const addButton = component.find('.add-button');
    addButton.simulate('click');
    expect(addButton.text()).toEqual("+ ADD MOVIE");
    expect(addButton.hasClass("add-button")).toEqual(true);
  });

});
