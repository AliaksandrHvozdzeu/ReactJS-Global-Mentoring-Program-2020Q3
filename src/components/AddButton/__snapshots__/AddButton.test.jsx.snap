// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when AddButton then snapshot created 1`] = `
<button
  className="add-button"
  onClick={[Function]}
  type="button"
>
  + ADD MOVIE
</button>
`;
